### PR TITLE
Added "Clear History" in settings - cleans recent searches 

### PR DESCRIPTION
**Pull request by EladKeyshawn**

### CODE REVIEW

**Comment by mhutti1** [comment]
Looks good. We just need to wipe each web-views history as well.

**Comment by kelson42** [comment]
@EladKeyshawn Can you do the clean history for _all_ webviews/tabs? Then we could probably merge.

**Comment by EladKeyshawn** [comment]
What do you mean each webview? Deleting all tabs?

**Comment by kelson42** [comment]
yes

**Comment by EladKeyshawn** [comment]
@kelson42 So once i click clean all history it cleans the recent searches and closes all tabs right?

File: android/res/values/strings.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
   <string name="go_to_play_store">Go to Play Store</string>
   <string name="no_bookmarks">No bookmarks!</string>
   <string name="menu_bookmarks_list">Bookmarks</string>
-  <string name="no_section_info">No content headers found</string>
-  <string name="request_storage">To access ZIM files we need access to your storage</string>
+  <string name="no_section_info">No Content Headers Found</string>
+  <string name="request_storage">To access zim files we need access to your storage</string>
+  <string name="history_dialog">Are you sure you want to delete all history?</string>
+  <string name="deleteRecentSearchItem">Delete this item?</string>
+  <string name="pref_clear_all_history_title">Clear all history</string>
```
Comment (mhutti1) [inline android/res/values/strings.xml]: Clear search history

File: android/res/values/strings.xml [hunk 2]
```diff
@@ -55,6 +55,11 @@
...
+  <string name="pref_clear_all_history_summary">Clears recent searches,tabs history</string>
```
Comment (mhutti1) [inline android/res/values/strings.xml:64]: Clears search history and resets active tabs

File: android/res/values/strings.xml [hunk 3]
```diff
@@ -55,6 +55,11 @@
   <string name="go_to_play_store">Go to Play Store</string>
   <string name="no_bookmarks">No bookmarks!</string>
   <string name="menu_bookmarks_list">Bookmarks</string>
-  <string name="no_section_info">No content headers found</string>
-  <string name="request_storage">To access ZIM files we need access to your storage</string>
+  <string name="no_section_info">No Content Headers Found</string>
+  <string name="request_storage">To access zim files we need access to your storage</string>
+  <string name="history_dialog">Are you sure you want to delete all history?</string>
```
Comment (mhutti1) [inline android/res/values/strings.xml]: This currently displays for both buttons but only makes sense for one. We need to give each one a separate text such as:
Are you sure you want to delete your search history?
Are you sure you want to delete your search history and reset all active tabs?

File: android/src/org/kiwix/kiwixmobile/settings/KiwixSettingsActivity.java [hunk 4]
```diff
@@ -184,6 +242,39 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
...
+
+    private void clearAllHistoryDialog() {
+      new AlertDialog.Builder(getActivity())
+          .setTitle("Clear All History")
+          .setMessage(getResources().getString(R.string.history_dialog))
+          .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int which) {
+              deleteSearchHistoryFromDb();
+              allHistoryCleared = true;
+              Toast.makeText(getActivity(), "All History Cleared", Toast.LENGTH_SHORT).show();
```
Comment (mhutti1) [inline android/src/org/kiwix/kiwixmobile/settings/KiwixSettingsActivity.java]: This text should be localised

File: android/src/org/kiwix/kiwixmobile/settings/KiwixSettingsActivity.java [hunk 5]
```diff
@@ -91,13 +123,39 @@ public void onCreate(Bundle savedInstanceState) {
...
 
+
+    private void recentSearchConfirmDialog() {
+      new AlertDialog.Builder(getActivity())
+          .setTitle("Clear History")
+          .setMessage(getResources().getString(R.string.history_dialog))
+          .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int which) {
+              deleteSearchHistoryFromDb();
+              Toast.makeText(getActivity(), "Recent searches cleared", Toast.LENGTH_SHORT).show();
```
Comment (mhutti1) [inline android/src/org/kiwix/kiwixmobile/settings/KiwixSettingsActivity.java]: This text should be localised

File: android/src/org/kiwix/kiwixmobile/SearchActivity.java [hunk 6]
```diff
@@ -117,4 +123,58 @@ private void sendMessage(String uri) {
...
+    return true;
+  }
+
+  private void deleteSpecificSearchDialog(final String search) {
+    new AlertDialog.Builder(this)
+        .setMessage(getResources().getString(R.string.deleteRecentSearchItem))
+        .setPositiveButton("Delete", new DialogInterface.OnClickListener() {
+          public void onClick(DialogInterface dialog, int which) {
+            deleteSpecificSearchItem(search);
+            Toast.makeText(getBaseContext(), "Recent search removed", Toast.LENGTH_SHORT).show();
```
Comment (mhutti1) [inline android/src/org/kiwix/kiwixmobile/SearchActivity.java]: This text should be localised

File: android/src/org/kiwix/kiwixmobile/KiwixMobileActivity.java [hunk 7]
```diff
@@ -579,13 +547,12 @@ private KiwixWebView restoreTabAtIndex(String url, int index) {
...
 
   private void closeTab(int index) {
-
     if (mWebViews.size() > 1) {
       if (mCurrentWebViewIndex == index) {
         if (mCurrentWebViewIndex >= 1) {
           selectTab(mCurrentWebViewIndex - 1);
           tempForUndo = mWebViews.get(index);
-
+          prevSize = mWebViews.size(); //****
```
Comment (mhutti1) [inline android/src/org/kiwix/kiwixmobile/KiwixMobileActivity.java]: What are the //***\* for?

File: android/res/values/strings.xml (hunk 3 above)
Comment (EladKeyshawn) [inline android/res/values/strings.xml]: It deletes all history and its pretty clear in the summary

Comment (mhutti1) [inline android/res/values/strings.xml]: But it asks if the user wants to delete **all** history when they only want to delete recent searches.

**Pull request merged**